apps/test/unit/studio/itemTest.js

```diff
@@ -11,32 +16,79 @@ import {
   BEHAVIOR_GRID_ALIGNED
 } from '@cdo/apps/studio/constants';
 
-import Item from '@cdo/apps/studio/Item';
-
 describe('item', () => {
   before(() => {
     Studio.trackedBehavior = {};
     Studio.trackedBehavior.createdItems = {};
+    Studio.SQUARE_SIZE = 50;
   });
 
   let item;
 
   describe('update', () => {
+    describe('if destination is not yet set', () => {
+      before(() => {
+        // the destination-setting logic uses _.shuffle to semi-randomize the
+        // set of possible destinations. Stub that with something deterministic
+        // to make testing easier
+        sinon.stub(_, 'shuffle').callsFake(ar => ar);
+      });
+
+      let targetSprite;
+      beforeEach(() => {
+        item = new Item({});
+        item.x = 100;
+        item.y = 100;
+
+        item.hasWall = sinon.stub().returns(false);
+
+        targetSprite = new Sprite({});
+        targetSprite.x = 200;
+        targetSprite.y = 200;
+        Studio.sprite = [targetSprite];
+        item.targetSpriteIndex = 0;
+      });
+
+      it('sets an arbitrary direction on wander', () => {
+        item.activity = BEHAVIOR_WANDER;
+        item.update();
+        expect(item.dir).to.equal(Direction.NORTH);
```

Comment:
Reading this test in-place, it's not totally clear why 'an arbitrary direction' always means `Direction.NORTH`.  Something I've been doing when I need to lock randomness like this in a test is to stub/restore a method in before/after, but set its fake return value (with an explanation) in the relevant test.

```js
beforeEach(() => sinon.stub(_, 'shuffle').callsThrough());
afterEach(() => _.shuffle.restore());

it('sets an arbitrary direction on wander', () => {
  // Comment explaining why we need to squelch randomness
  _.shuffle.callsFake(ar => ar);
  // ...rest of test case
});
```